extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/handler/TradfriLightHandler.java

```diff
@@ -0,0 +1,322 @@
+        data.setOnOffState(onOff == OnOffType.ON);
+        set(data.getJsonString());
+    }
+
+    private void setColorTemperature(PercentType percent) {
+        LightData data = new LightData();
+        data.setColorTemperature(percent).setTransitionTime(DEFAULT_DIMMER_TRANSITION_TIME);
+        set(data.getJsonString());
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            logger.debug("Refreshing channel {}", channelUID);
+            coapClient.asyncGet(this);
+            return;
+        }
+
+        switch (channelUID.getId()) {
+            case CHANNEL_BRIGHTNESS:
+                if (command instanceof PercentType) {
+                    setBrightness((PercentType) command);
+                } else if (command instanceof OnOffType) {
+                    setState(((OnOffType) command));
+                } else if (command instanceof IncreaseDecreaseType) {
+                    if (state != null) {
+                        int current = state.getBrightness().intValue();
+                        if (IncreaseDecreaseType.INCREASE.equals(command)) {
+                            setBrightness(new PercentType(Math.min(current + STEP, PercentType.HUNDRED.intValue())));
+                        } else {
+                            setBrightness(new PercentType(Math.max(current - STEP, PercentType.ZERO.intValue())));
+                        }
+                    } else {
+                        logger.debug("Cannot handle inc/dec as current state is not known.");
+                    }
+                } else {
+                    logger.debug("Cannot handle command {} for channel {}", command, channelUID);
+                }
+                break;
+            case CHANNEL_COLOR_TEMPERATURE:
+                if (command instanceof PercentType) {
+                    setColorTemperature((PercentType) command);
+                } else if (command instanceof IncreaseDecreaseType) {
+                    if (state != null) {
+                        int current = state.getColorTemperature().intValue();
+                        if (IncreaseDecreaseType.INCREASE.equals(command)) {
+                            setColorTemperature(
+                                    new PercentType(Math.min(current + STEP, PercentType.HUNDRED.intValue())));
+                        } else {
+                            setColorTemperature(new PercentType(Math.max(current - STEP, PercentType.ZERO.intValue())));
+                        }
+                    } else {
+                        logger.debug("Cannot handle inc/dec as current state is not known.");
+                    }
+                } else {
+                    logger.debug("Can't handle command {} on channel {}", command, channelUID);
+                }
+                break;
+            default:
+                logger.error("Unknown channel UID {}", channelUID);
```

Comment:
if we get here, it would indeed be a severe bug in the framework, so imho ok to log as error.